### PR TITLE
[AAP-75136] feat(ci): add Codecov coverage upload workflow

### PR DESCRIPTION
**Pull request by john-westcott-iv**
## Description

- Adds a `workflow_run`-triggered GitHub Actions workflow that uploads coverage to Codecov.
  Fires after every Unit Tests completion — both PR runs (for diff coverage comments) and
  devel merges (to establish the baseline that PR diffs are measured against).
- Adds `codecov.yml` configuration with coverage exclusions matching SonarCloud's
  `sonar.coverage.exclusions` and a `unit-tests` flag for flag analytics.
- Follows the same artifact-download pattern as the sonar-pr.yml workflow.

## Type of Change
- [x] New feature (non-breaking change which adds functionality)

## Self-Review Checklist
- [x] I have performed a self-review of my code
- [x] I have added relevant comments to complex code sections
- [x] I have considered the security impact of these changes
- [x] I have considered performance implications
- [x] I have thought about error handling and edge cases

### Prerequisites
- CODECOV_TOKEN must be set as a GitHub Actions secret (done)
- Codecov GitHub App must be installed on the repo (done)

### Steps to Test
1. Merge this PR to devel
2. Observe the Codecov workflow triggers after Unit Tests complete on devel
3. Open a new PR and confirm Codecov posts a coverage summary comment

### Expected Results
- Coverage appears in Codecov dashboard under ansible/jewel
- PR coverage comments show patch and project coverage
- Flag analytics shows the `unit-tests` flag in the Codecov Flags tab

### Intent Adherence: 85/100

| Criterion | Score |
|-----------|-------|
| Solves the problem described in the story | 40/40 |
| No deviations from stated intent | 20/20 |
| No over-engineering or scope creep | 15/20 |
| A reviewer would agree the code matches the story | 10/20 |

### Required Actions
- [ ] Requires coordination with other teams
  COVERPORT-105 approval for public repo onboarding
- [ ] Enable flag analytics in Codecov UI (Flags tab) after first upload

Assisted-by: Claude Code / Opus 4.6 (Anthropic)

## Summary by CodeRabbit

* **Chores**
  * CI now collects unit-test coverage artifacts after successful test runs and automatically uploads them to Codecov, helping track coverage over time and fail CI on upload errors.
  * Added Codecov configuration to control which files are measured, set coverage targets to automatic, and enable a dedicated "unit-tests" coverage flag for clearer, carry-forward coverage reporting.

### CODE REVIEW

File: .github/workflows/codecov.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      actions: read
+    if: >-
+      github.event.workflow_run.conclusion == 'success' &&
+      (github.repository == 'ansible/jewel' || github.repository == 'ansible/jewel-debug')
+    steps:
+      - uses: actions/checkout@v5
+        with:
+          ref: ${{ github.event.workflow_run.head_sha }}
+
+      - name: Download coverage artifacts
```
[the remark this answers is not in the document]
Comment (john-westcott-iv) [reply .github/workflows/codecov.yml]: Fixed in b97cbb31 — removed the actions/checkout step entirely. The workflow only needs the downloaded coverage artifacts, not the source tree.

File: .github/workflows/codecov.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+          echo "file_list=$csv" >> "$GITHUB_OUTPUT"
+
+          first=$(echo "$files" | head -1)
+          pr=$(grep -m 1 '<!-- PR' "$first" 2>/dev/null | awk '{print $3}' || true)
+          echo "pr_number=$pr" >> "$GITHUB_OUTPUT"
+          if [ -n "$pr" ]; then
+            echo "Detected PR #$pr from coverage artifact"
+          else
+            echo "No PR number found (branch push)"
+          fi
```
[the remark this answers is not in the document]
Comment (john-westcott-iv) [reply .github/workflows/codecov.yml:62]: Fixed in b97cbb31 — replaced fragile awk with grep -oP and now loops through all coverage files instead of only checking the first one.

**Review by huffmanca — APPROVED** [review]
Tentatively approving this one, as I don't see any issues with the workflow.
